Makefile

```diff
@@ -16,4 +18,40 @@ install-hooks: venv
 	venv/bin/pre-commit install -f --install-hooks
 
 vendor: Puppetfile
-	r10k puppetfile install
+	r10k puppetfile install --verbose --color
+
+.PHONY: octocatalog-diff-uncached
+$(WORKSPACE)/.octocatalog-diff-cache:
+octocatalog-diff-uncached:
+	octocatalog-diff \
+		--bootstrap-then-exit \
+		--bootstrapped-from-dir=$(WORKSPACE)/.octocatalog-diff-cache
+
+# Run octocatalog-diff for a particular hostname
+# Add a --debug flag to get much more verbose output
+diff_%: $(WORKSPACE)/.octocatalog-diff-cache
+	octocatalog-diff \
+		-n $*.ocf.berkeley.edu \
+		--bootstrapped-to-dir=$(WORKSPACE)/.octocatalog-diff-cache \
+		--enc-override environment=production,parameters::use_private_share=false \
+		--ignore 'Ini_setting[puppet.conf/master/storeconfigs]' \
+		--ignore 'Ini_setting[puppet.conf/master/storeconfigs_backend]' \
+		--ignore 'Ini_setting[puppetdbserver_urls]' \
+		--ignore 'Ini_setting[soft_write_failure]' \
+		--ignore 'File[/tmp/*/routes.yaml]' \
+		--display-detail-add
+
+# Run octocatalog-diff across all nodes that can be fetched from puppetdb
+# TODO: Make this faster by just selecting a single node from each class we
+# care about (not selecting all desktops/hozers for example)
+#all_diffs: $(WORKSPACE)/.octocatalog-diff-cache
+all_diffs: octocatalog-diff-uncached
+	curl -s --tlsv1 \
+		--cacert /etc/ocfweb/puppet-certs/puppet-ca.pem \
+		--cert /etc/ocfweb/puppet-certs/puppet-cert.pem \
+		--key /etc/ocfweb/puppet-certs/puppet-private.pem \
+		https://puppetdb:8081/pdb/query/v4/nodes \
+	| jq -r '.[] | .certname' \
+	| cut -d '.' -f 1 \
```

Comment:
Is this cutting out the `.ocf.berkeley.edu`? Why not just leave it in, since you're adding it again in the `diff_%` rule?

Reply:
Heh, yeah that'd be better, I just cut it out since I was running it as `make diff_supernova` and wanted to match that, but there isn't really much need to split out just the hostname and then include it again like this

Reply:
In practice I think doing it this way would be about the same complexity, since I'd like to keep a nice way to run individual diffs per host (the `make diff_supernova` interface for instance). However, this means stripping and re-adding the hostname beforehand unless there's a new make target like `diff_fqdn_%` to take in fqdns and then `diff_%` just passes onto that one (this works too, but looks about the same in terms of complexity to me). Thoughts?

Reply:
If you expect humans to actually run that `diff` command then feel free to keep it the way it is-- I just thought this would always be automated, in which case it's better to be explicit. It's not worth adding more Makefile rules for this.

Reply:
Yeah, I didn't make it clear that it's ready for humans to use, so I'll add a comment about that or explaining why the command is set up like it is and not using the fqdn